### ISSUE TITLE
Feature: nadam optimizer 

### ISSUE THREAD

**Issue by cancan101** [tensorflow/tensorflow#7715]
Adam with nesterov momentum. 

Keras has it: https://keras.io/optimizers/#nadam
TF implementation (old and doesn't quite work) : https://github.com/tdozat/Optimization/blob/master/tensorflow/nadam.py

**Comment by inflation** [comment]
I'm thinking about the same approach like `MomentumOptimizer` with a `use_nesterov` flag. But I don't know if it is still preferable to change the interface now.

**Comment by poxvoculi** [comment]
Is there a reason this shouldn't simply be a community supplied option, like the cited version?   Are you suggesting a specific modification to TF?  Otherwise perhaps it would be better to discuss such an extension on stack overflow.

**Comment by cancan101** [comment]
@poxvoculi Do you think it makes sense to add a `use_nesterov` flag into Adam (in the tf repos)?

**Comment by poxvoculi** [comment]
@ebrevdo: who would have an informed opinion on this?

**Comment by inflation** [comment]
@poxvoculi A C++ implementation would be faster than python hacks like above. Also, maybe an awaiting googler label?

**Comment by ebrevdo** [comment]
Paul: probably vincent would know how popular / useful this optimizer is
for the general case.

**Comment by vincentvanhoucke** [comment]
Plausible. I have no experience with it. If someone implements it, it sounds like something we would incorporate.

**Comment by girving** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
